sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala

```diff
@@ -610,15 +610,27 @@ private[sql] object QueryExecutionErrors extends QueryErrorsBase with ExecutionE
   }
 
   def binaryArithmeticCauseOverflowError(
-      eval1: Short, symbol: String, eval2: Short): SparkArithmeticException = {
-    new SparkArithmeticException(
-      errorClass = "BINARY_ARITHMETIC_OVERFLOW",
-      messageParameters = Map(
-        "value1" -> toSQLValue(eval1, ShortType),
-        "symbol" -> symbol,
-        "value2" -> toSQLValue(eval2, ShortType)),
-      context = Array.empty,
-      summary = "")
+      eval1: Short, symbol: String, eval2: Short, hint: String): SparkArithmeticException = {
+    if (hint == "") {
```

Comment:
Could you split the function to two separate methods w/ and w/o suggested function. In this way you can avoid checking the argument, and eliminate `if ... else` at all/
```scala
def binaryArithmeticCauseOverflowError( eval1: Short, symbol: String, eval2: Short): SparkArithmeticException

def binaryArithmeticCauseOverflowError( eval1: Short, symbol: String, eval2: Short, suggestedFunc: String): SparkArithmeticException
```